corehq/apps/app_manager/xform.py

```diff
@@ -2133,7 +2133,7 @@ def infer_vellum_type(control, bind):
 
 def find_missing_instances(form_source):
     from corehq.apps.app_manager.suite_xml.post_process.instances import get_all_instances_referenced_in_xpaths
-    instance_declaration_re = r"""<instance.*id=\"([\w\-:]+)\""""
+    instance_declaration_re = r"""<instance[^>]*id=\"([\w\-:]+)\""""
```

Comment:
Also, using a regexp on the entire xform seems less than ideal (although maybe not a huge concern). This will pick up things that look like instance definitions in parts of the xml where instance definitions should not be found. Is the XML parsed prior to rendering? If yes, wouldn't it be better to parse instances using a real XML parser that knows about things like document structure?